privacyidea/lib/tokens/hotptoken.py

```diff
@@ -255,10 +256,16 @@ def get_init_detail(self, params=None, user=None):
                                                   "img": create_img(oath_url,
                                                                     width=250)
                                                   }
+                                                 
+                    key_bin = binascii.unhexlify(otpkey)
```

Comment:
This block (line 260-264) did not get executed for me, because the ``cr_oath`` failed with an exception. So in my special case this block was not called. 
I recommend moving this block up to line 224 - before ``cr_google``.